### PR TITLE
Avoid GET_INVENTORY on every tick

### PR DESCRIPTION
**Pull request by vosk**
## Short Description:

Currently transfer_pokemon and recycle items do an inventory refresh on every run. Instead of this, rely on previously updated inventory and if changes are made, trigger a refresh

## Fixes (provide links to github issues if you can):
-
-
-

### CODE REVIEW

File: pokemongo_bot/cell_workers/recycle_items.py [hunk 1]
```diff
@@ -89,6 +88,10 @@ def work(self):
                 if self.item_should_be_recycled(item_in_inventory, amount_to_recycle):
                     if ItemRecycler(self.bot, item_in_inventory, amount_to_recycle).work() == WorkerResult.ERROR:
                         worker_result = WorkerResult.ERROR
+                    else:
+                        should_update_inventory=True
```
Comment (BriceSD) [inline pokemongo_bot/cell_workers/recycle_items.py]: The ItemRecycler already updates the inventory after discarding items

File: pokemongo_bot/cell_workers/recycle_items.py [hunk 2]
```diff
@@ -77,9 +77,8 @@ def work(self):
         :rtype: WorkerResult
         """
         # TODO: Use new inventory everywhere and then remove the inventory update
-        # Updating inventory
-        inventory.refresh_inventory()
```
Comment (BriceSD) [inline pokemongo_bot/cell_workers/recycle_items.py]: This refresh_inventory call is to make sure that this worker will try to recycle items that actually are in the inventory. We need it until every task update the inventory when looting/using an item.

**Comment by BriceSD** [comment]
@vosk Afaik we’re trying to get rid of inventory api related calls. Thus having only one init_inventory() an never using refresh_inventory(), but keeping track of inventory changes manually.

**Comment by TheSavior** [comment]
Closing this because as @BriceSD said, this isn't the approach that makes sense for the bot. We really want to rely on the inventory classes more, perhaps you guys could figure out how to continue the push in the right direction.

**Pull request closed without merging**